packages/components/src/date-time/date.js

```diff
@@ -1,24 +1,136 @@
 /**
  * External dependencies
  */
-import ReactDatePicker from 'react-datepicker';
+import classnames from 'classnames';
 import moment from 'moment';
+import { DayPickerSingleDateController } from 'react-dates';
+
+/**
+ * WordPress dependencies
+ */
+import { withInstanceId } from '@wordpress/compose';
+import { __ } from '@wordpress/i18n';
+import { Component } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import Button from '../button';
 
 /**
  * Module Constants
  */
 const TIMEZONELESS_FORMAT = 'YYYY-MM-DDTHH:mm:ss';
 
-function DatePicker( { currentDate, onChange, ...args } ) {
-	const momentDate = currentDate ? moment( currentDate ) : moment();
-	const onChangeMoment = ( newDate ) => onChange( newDate.format( TIMEZONELESS_FORMAT ) );
+class DatePicker extends Component {
+	constructor() {
+		super( ...arguments );
+
+		this.state = {
+			descriptionVisible: false,
+		};
+
+		this.onChangeMoment = this.onChangeMoment.bind( this );
+		this.onClickDescriptionToggle = this.onClickDescriptionToggle.bind( this );
+	}
+
+	onChangeMoment( newDate ) {
+		const { currentDate, onChange } = this.props;
+
+		const momentDate = currentDate ? moment( currentDate ) : moment();
+		const momentTime = {
+			seconds: momentDate.seconds(), minutes: momentDate.minutes(),
+			hours: momentDate.hours(),
+		};
+
+		onChange( newDate.set( momentTime ).format( TIMEZONELESS_FORMAT ) );
+	}
+
+	onClickDescriptionToggle() {
+		this.setState( { descriptionVisible: ! this.state.descriptionVisible } );
```

Comment:
Oh my, I understand what the issue is now. This is help for the entire component but is rendered inside the `DatePicker`, leaving the `TimePicker` free to render regardless of the state of help text which… argh. I'll probably need to refactor this component a bit to make it work.

Reply:
And if I move things around then the `DatePicker` alone loses its help text.

Maybe just cramming a focus lock onto the help text, if rendered, will do the trick. I'll try that...

Reply:
Works now. 👍 

Seems an alternative would have been to just have a single button for the toggle, always rendered, rather than separate faux-link "Calendar Help" (right-aligned) and the button "Close", with added benefits of: focus doesn't move (because it's the same element) and we have a consistent display for the title.

Reply:
I am happy to do that in a follow-up PR. I just want to get this merged 😆

Reply:
>I am happy to do that in a follow-up PR. I just want to get this merged 😆

Where can I track the progress of this follow-up task?

Reply:
Oops, hah, tab was left open but I never clicked file. #10621